src/sds_data_model/table.py

```diff
@@ -0,0 +1,204 @@
+from dataclasses import dataclass
+import inspect
+import io
+import json
+from logging import INFO, info, basicConfig
+import requests
+from typing import Any, Dict, List, Optional, TypeVar, Union
+
+from pandas import (DataFrame, Series,
+                    read_csv, read_json, read_excel, read_parquet,
+                    to_numeric)
+from pathlib import Path
+from urllib.parse import urlparse
+
+from sds_data_model.metadata import Metadata
+
+basicConfig(format="%(levelname)s:%(asctime)s:%(message)s", level=INFO)
+
+_TableLayer = TypeVar("_TableLayer", bound="TableLayer")
+
+
+def _find_closing_bracket(string):
+    """Find the index of the closing bracket in a string."""
+    flag = 1
+    string_index = 1
+    while flag > 0:
+        if string[string_index] == "(":
+            flag += 1
+        elif string[string_index] == ")":
+            flag -= 1
+        string_index += 1
+    return string_index - 1
+
+
+def _get_function_input(func_name: str, frame: inspect.types.FrameType) -> str:
+    """Get the input code arguments to a function as a string."""
+    code_input = inspect.getouterframes(frame, 1)
+    code_input = [f for f in code_input
+                  if f'{func_name}(' in f.code_context[0]]
+    code_input = code_input[0].code_context[0]
+    code_input = code_input[len(func_name) + code_input.find(f'{func_name}('):]
+    code_input = code_input[1:_find_closing_bracket(code_input)]
+    return code_input
```

Comment:
As you suspected, you can do this in less lines with regex. 

```suggestion
def _get_function_input(func_name: str, frame: inspect.types.FrameType) -> str:
    """Get the input code arguments to a function as a string."""
    code_input = inspect.getouterframes(frame, 1)
    code_contexts = tuple(frame.code_context for frame in code_input)
    code_context_string = code_contexts[-1][0]
    function_call_strings = (
        search(fr"\.{func_name}\(([\w|\W]+?\))", code_context_string)
        .group(1)
    )
    return function_call_strings
```

This doesn't solve the chaining problem though...

Reply:
The logging-related functions can be removed now the logic has been moved to #31

Reply:
All logging removed from current code, to be reinstated with new logging decorator in new issue.